core/src/test/java/greencity/service/impl/GoalServiceImplTest.java

```diff
@@ -41,4 +65,37 @@ public void findAllTest() {
         when(goalTranslationRepo.findAllByLanguageCode(AppConstant.DEFAULT_LANGUAGE_CODE)).thenReturn(goalTranslations);
         assertEquals(goalService.findAll(AppConstant.DEFAULT_LANGUAGE_CODE), goalsDto);
     }
+
+    @Test
+    void getUserGoalResponseDtoFromPredefinedGoal() {
+        when(modelMapper.map(predefinedUserGoal, UserGoalResponseDto.class)).thenReturn(predefinedUserGoalDto);
+        when(languageService.extractLanguageCodeFromRequest()).thenReturn(ModelUtils.getLanguage().getCode());
+        Goal goal = predefinedUserGoal.getGoal();
+        when(goalRepo.findById(anyLong())).thenReturn(Optional.ofNullable(goal));
+        when(goalTranslationRepo.findByGoalAndLanguageCode(any(Goal.class), anyString())).thenReturn(Optional.of(ModelUtils.getGoalTranslation()));
+        assertEquals(predefinedUserGoalDto, goalService.getUserGoalResponseDtoFromPredefinedGoal(predefinedUserGoal));
+    }
+
+    @Test
+    void getUserGoalResponseDtoFromPredefinedGoalFailed() {
+        when(modelMapper.map(predefinedUserGoal, UserGoalResponseDto.class)).thenReturn(predefinedUserGoalDto);
+        when(languageService.extractLanguageCodeFromRequest()).thenReturn(ModelUtils.getLanguage().getCode());
+        when(goalRepo.findById(anyLong())).thenReturn(Optional.empty());
+        assertThrows(GoalNotFoundException.class, () -> goalService.getUserGoalResponseDtoFromPredefinedGoal(predefinedUserGoal));
+    }
+
+    @Test
+    void getUserGoalResponseDtoFromCustomGoal() {
+        when(modelMapper.map(customUserGoal, UserGoalResponseDto.class)).thenReturn(customUserGoalDto);
+        CustomGoal customGoal = customUserGoal.getCustomGoal();
+        when(customGoalRepo.findById(anyLong())).thenReturn(Optional.ofNullable(customGoal));
+        assertEquals(customUserGoalDto, goalService.getUserGoalResponseDtoFromCustomGoal(customUserGoal));
```

Comment:
Statement 

`goalService.getUserGoalResponseDtoFromCustomGoal(customUserGoal)`

should be on a separate line with blank lines above and below. The same for each test.